.github/workflows/cacti-build-and-test.yaml

```diff
@@ -0,0 +1,54 @@
+# Unified Cacti Build and Test
+---
+env:
+  NODEJS_VERSION: v18.18.2
+  
+jobs:
+
+  build-dev:
+    continue-on-error: false
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4.1.1
+      - uses: ./.github/actions/build-and-cache/
+
+  yarn-tasks:
+    name: yarn ${{ matrix.yarn-task }}
+    needs: build-dev
+    runs-on: ubuntu-22.04
+    continue-on-error: true
+    strategy:
+      matrix:
+        yarn-task: 
+        - tools:validate-bundle-names
+        - custom-checks
+        - lint
+    steps:
+      - uses: actions/checkout@v4.1.1
+      - uses: ./.github/actions/build-and-cache/
+      - run: yarn ${{ matrix.yarn-task }}
+      
+  package-test:
+    name: test ${{ matrix.package-to-test }}
+    needs: build-dev
+    runs-on: ubuntu-22.04
+    continue-on-error: true
+    strategy:
+      matrix:
+        # this can eventually be the output of npx 'lerna changed' or 'lerna list', depending on how invoked
+        package-to-test: 
+        - cactus-api-client
+        - cactus-verifier-client
+        - cactus-test-plugin-consortium-manual
+    steps:
+      - uses: actions/checkout@v1
+      - uses: ./.github/actions/build-and-cache/ # restore cache and build what's not cached
+      - name: Test
+        run: npx lerna run test --scope "@hyperledger/${{ matrix.package-to-test }}" --stream
+
+name: Unified Cacti Build and Test
```

Comment:
@jenniferlianne Please use a much shorter name like `CI` - Normally I'd agree about being descriptive, but unfortunately GitHub's UI punishes the user for long action/job/workflow names because they get cropped off with no way to scroll and this makes it very tedious to find out sometimes which job is failing to the point that you have to literally go and inspect the HTML markup just to find out which job is failing.